Automattic/wp-calypso
# Settings: Remove FormToggle in favor of ToggleControl

#### Changes proposed in this Pull Request

In #52915 we removed the last real need for a `FormToggle` component in Calypso. With that merged, `FormToggle` became a simple proxy of the `ToggleControl` component, and the only real difference is the way that the label is provided - in `FormToggle` we provide those as `children`, while in `ToggleControl` that's being done with the `label` prop. So `FormToggle` can just be removed in favor of `ToggleControl`, there is no real need to maintain that abstraction anymore.

This PR updates all of the `FormToggle` instances in the site settings to use `ToggleControl`. While it may look like a big PR, this is 99% because **we're now passing the `children` as a `label` prop.**

This PR is split off #52917.

#### Testing instructions

* Verify all tests pass.
* Verify we correctly replace all `FormToggle` `children` props with `label` prop when using `ToggleControl`. 
* Going through all settings tabs in /settings/general/:site, verify all toggles work and look the same way as before, where `:site` is a:
  * Jetpack site
  * Atomic site
  * WP.com simple site

#### Notes

We're also fixing a couple of minor spacing discrepancies in the meantime.
